File: vbnet/resistor-color-duo/ResistorColorDuo.vb
Imports System

Public Module ResistorColorDuo
    Dim lst() As String = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

    Public Function Value(colors As String) As Integer
        Return Integer.Parse(Array.IndexOf(colorList, colors(0)).ToString() & Array.IndexOf(colorList, colors(1)).ToString())
    End Function

End Module

File: euphoria/resistor-color-duo/resistor-color-duo.ex
sequence RESISTOR_COLORS = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

public function value(sequence colors)
    return (find(colors[1], RESISTOR_COLORS) - 1) * 10 + find(colors[2], RESISTOR_COLORS) - 1
end function
